src/output.rs

```diff
@@ -17,10 +18,14 @@ pub enum Argument {
 }
 
 impl FilterPrototype {
-    pub fn new(name: &str, arguments: Vec<Argument>) -> FilterPrototype {
+    pub fn new(name: &str,
+               arguments: Vec<Argument>,
+               named_arguments: Vec<(String, Argument)>)
+               -> FilterPrototype {
         FilterPrototype {
             name: name.to_owned(),
             arguments: arguments,
+            named_arguments: named_arguments,
```

Comment:
>  What exactly are the semantics for named arguments that we need to support, like mixing positional and named?
> At first I started by only supporting named args at the end of the filter, but then I had the brilliant idea of checking how Shopify/liquid does it. As you can see, named args and positional args can be mixed.

Some things we need to consider
- When mixing named and positional, how does the position index change?  If I do `val0, arg1: val1, valN`, is `valN` `val1` or `val2`?
- Repeating a named argument: `arg0: val0, arg0: something`
- Repeating a positional argument as a named argument: `val0, val1, val2, arg0: something`